### DISCUSSION TITLE
pytest_configure: injecting variables into pytest namespace is mypy-incompatible with pytest 8

### DISCUSSION THREAD

**Discussion by sigma67** [pytest-dev/pytest#11881] [General] [answered]
Hello, thanks for your great and invaluable work on pytest. I just updated to pytest 8 and I'm getting linter errors on the following scenario: 

https://docs.pytest.org/en/latest/deprecations.html#pytest-namespace

```python
import pytest


def pytest_configure():
    pytest.my_symbol = MySymbol()
```

The error:

```
error: Module has no attribute "my_symbol" [attr-defined]
```

As this approach is mentioned in the docs I'd expect it to still work. 

- Is there a way to still do this without disabling mypy everywhere the variable is used? 
- Is there a different approach that is considered better? 
 
Context: I'm using the variable to share data between a pytest plugin I develop and the module (i.e. module defines the variable - plugin uses it).

`pytest 8.0.0, mypy 1.8.0`

- [x] a detailed description of the bug or problem you are having
- [x] output of `pip list` from the virtual environment you are using
- [x] pytest and operating system versions
- [x] minimal example if possible

**Comment by RonnyPfannschmidt** [comment, marked as answer]
ever since the pytest_namespace hook has been deprecated and subsequently been removed, the suggesion is to have own symbols in own modules

pytest owns the pytest namespace - nothing else - make a own namespace to store own symbols

**Reply by sigma67** [reply]
Ok, thanks for the fast reply and the hint 👍
